java/arcs/core/data/testdata/WriterReaderExample.arcs

```diff
@@ -24,3 +24,5 @@ recipe Consumption
   data: map 'my-handle-id'
   Reader
     data: reads data
+
+// TODO: add more tests for ephemeral Arcs and stores with CreateableStorageKey.
```

Comment:
Can we associate this with a bug?

Reply:
actually, i think i'll have to do it in this PR, because it seems that some Allocator tests are broken with this exact use case. i'm investigating.

Reply:
added an additional recipe and a fix. tests passing now.